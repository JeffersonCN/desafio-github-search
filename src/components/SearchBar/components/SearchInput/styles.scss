    .searchInput {
        box-sizing: border-box;
        font-size: 20px;
        font-family: "Raleway";
        font-weight: 300;
        flex-grow: 2;
        padding-left: 15px;
        overflow: hidden;
        height: 50px;
        border: 0;
        border-radius: 2px 0 0 2px;
        box-shadow: 0 0 4px 0 rgba(0, 0, 0, 0.4);
        color: #5c5c5c;
        background-color: #fff;
    }