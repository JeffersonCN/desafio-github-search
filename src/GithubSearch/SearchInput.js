import React, { Component } from 'react';
import { colors, boxShadow } from '../styles';

const searchInputStyle = {
    width: "100%",
    height: "50px",
    boxSizing: "border-box",
    border: "1px solid #D8D8D8",
    borderRadius: "2px 0 0 2px",
    boxShadow: boxShadow,
    color: colors.gray,
    backgroundColor: colors.white,
    float: "left",
    paddingLeft: "17px",
    fontSize: "20px",
    fontFamily: "Raleway",
    fontWeight: "300",
}

class SearchInput extends Component {
    render() {
        return (
            <input type="text" className="SearchInput" style={searchInputStyle}/>
        );
    }
}

export default SearchInput;