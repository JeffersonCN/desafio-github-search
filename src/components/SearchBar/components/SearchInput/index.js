import React, { Component } from 'react';

import './styles.scss';

class SearchInput extends Component {
    render() {
        return (
            <input type="text" className="search-input"/>
        );
    }
}

export default SearchInput;