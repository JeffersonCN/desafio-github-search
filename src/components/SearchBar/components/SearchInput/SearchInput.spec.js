import React from 'react';
import { shallow } from 'enzyme';
import SearchInput from './';

describe('Testing SearchInput Component', () => {
    it('shoud render correctly', () => {
        const wrapper = shallow(
            <SearchInput />
        );

        expect(wrapper).toMatchSnapshot();
    });
});