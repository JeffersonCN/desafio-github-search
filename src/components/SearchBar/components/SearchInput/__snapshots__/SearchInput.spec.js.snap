// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SearchInput Component shoud render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "onChange": undefined,
      "onKeyUp": undefined,
      "type": "text",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "onChange": undefined,
        "onKeyUp": undefined,
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
